cli/popper/parser.py

```diff
@@ -387,3 +387,27 @@ def find_root_recursively(workflow, action, required_actions):
             workflow.action.pop(a)
 
         return workflow
+
+    @staticmethod
+    def parse_substitutions(wf, substitutions):
+
+        workflow = deepcopy(wf)
+        substitution_dict = dict()
+
+        for args in substitutions:
+            item = args.split('=')
+            substitution_dict[item[0]] = item[1]
+
+        for key in workflow.parsed_workflow['action']:
+            for var in substitution_dict:
+                uses_str = workflow.parsed_workflow['action'][key]['uses']
```

Comment:
the goal is to support substitutions on all parts of the workflow, not only `uses` attribute of actions.